### PR TITLE
highlight the implementation-specific stuff in aggregates chapter

### PR DESCRIPTION
**Pull request by hjwp**

### CODE REVIEW

File: chapter_07_aggregate.asciidoc [hunk 1]
```diff
@@ -787,7 +805,7 @@ read1, write1, read2, write2(succeed)
 //TODO maybe better diagrams here?
 
 In our simple case, it's not obvious which to prefer.  In a more complex
-scenario, `SELECT FOR UPDATE` might lead to more deadlocks, while `SERIALIZABLE`
+scenario, `SELECT FOR UPDATE` might lead to more deadlocks, while `READ REPETABLE`
```
Comment (xtaje) [inline chapter_07_aggregate.asciidoc]:
```suggestion
scenario, `SELECT FOR UPDATE` might lead to more deadlocks, while `REPEATABLE READ`
```

File: chapter_07_aggregate.asciidoc [hunk 2]
```diff
@@ -731,26 +749,26 @@ def test_concurrent_updates_to_version_are_not_allowed(postgres_session_factory)
...
-    isolation_level="SERIALIZABLE",
+    isolation_level="REPEATABLE READ",
 ))
 ----
 ====
 
 Transaction isolation levels are tricky stuff, it's worth spending time
-understanding https://www.postgresql.org/docs/9.6/transaction-iso.html[the
+understanding https://www.postgresql.org/docs/12/transaction-iso.html[the
 documentation].
```
Comment (xtaje) [inline chapter_07_aggregate.asciidoc]: I would make the caveat here about "spending time understanding your vendor's specific interpretation of isolation levels".  A specific example is that Oracle `SERIALIAZABLE` is equivalent to Postgres `REPEATABLE READ`.

Comment (hjwp) [reply chapter_07_aggregate.asciidoc]: will do.

one of the nice things in python is that we've decided that the `one --and preferably only one-- obvious way to do it` is Postgres :)

Comment (hjwp) [inline chapter_07_aggregate.asciidoc]:
```suggestion
Postgres documentation] (or whichever database you use--annoyingly, they all have quite radically different definitions.  Oracle's SERIALIZABLE is equivalent to Postgres's REPETABLE READ, for example.
```

File: chapter_07_aggregate.asciidoc [hunk 3]
```diff
@@ -648,10 +662,13 @@ concurrent attempts to do allocation against the same `Product`, one of them
 should fail, because they can't both update the version number.
 
 First let's simulate a "slow" transaction using a function that does
-allocation, and then does an explicit sleep:
+allocation, and then does an explicit sleep:footnote:[`time.sleep()` works well
+in our use case, but it's not the most reliable or efficient way to reproduce
+concurrency bugs.  Consider using semaphores or similar concurrency primitives
```
Comment (xtaje) [inline chapter_07_aggregate.asciidoc]:
```suggestion
concurrency bugs.  Consider using semaphores or similar synchronization primitives
```

**Pull request merged**